napi.h

```diff
@@ -355,6 +357,20 @@ namespace Napi {
     /// be needed to store this BigInt (i.e. the return value of `WordCount()`).
     void ToWords(int* sign_bit, size_t* word_count, uint64_t* words);
   };
+
+  /// A JavaScript date value.
+  class Date : public Value {
+  public:
+    static Date New(
+      napi_env env, ///< N-API environment
+      double value ///< Number value
+    );
+
+    Date();                               ///< Creates a new _empty_ Date instance.
+    Date(napi_env env, napi_value value); ///< Wraps a N-API value primitive.
+
+    double NumberValue() const;   ///< Converts a Date value to a number value.
```

Comment:
I'm not sure about the name of this method maybe we could consider  other altrenatives like `TimeValue` or `DateValue` or `DoubleValue` what do you think?

Reply:
I thought of V8's `Value::NumberValue()` method. We could also use `ValueOf()` like V8's [Date](https://v8docs.nodesource.com/node-10.6/d4/df7/classv8_1_1_date.html#a24c1e040bf6c6e9623b3d8867e622d6e) class or just a complete different name.

Reply:
From discussion in the N-API team meeting we suggest:

ToNumber()  - similar to some of our other conversion functions and similar to how casting to a Number does conversion on JavaScript side.

Reply:
Thanks @mhdawson. I am fine with every naming but ToNumber() makes me think I will get a Napi::Number and not a double primitive. Napi::Number has a DoubleValue() method. Not really consistent isn't it?

Reply:
@mhdawson @NickNaso So really `ToNumber()`? In the codebase this is currently used to cast to `Napi::Number` and not to a double primitive. I would like to see `DoubleValue` because it is already used in the codebase or [Date.valueOf()](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Date/valueOf) referring to the JavaScript Date object.

Reply:
Hi @mathiask88 @mhdawson,
I think that valueOf is a good choice like reported by @mathiask88 it's part of the standard and it's used even in module like **[momentjs](https://momentjs.com/docs/#/displaying/unix-timestamp-milliseconds/)**. 
Furthermore I searched how this is handled in other platforms / languages and above I report what I found:
- CurrentTimeMillis
- GetTimeInMillis 
- Millis 
- CurrentMillis
At this time I think that the best option is **valueOf**. 
Could someone help and give a feedback? @nodejs/n-api

Reply:
Any news on this?

Reply:
I'm ok with valueOf as well.